<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="GazeboSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="arm_joint">
            <command_interface name="effort"/>
            
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="pendulum_joint">
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

    </ros2_control>

    <gazebo>
        <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
            <parameters>$(find fura2a)/config/controller.yaml</parameters>
        </plugin>
    </gazebo>

    <gazebo>
    <plugin name="pendulum_disturbance" filename="libgazebo_ros_joint_effort.so">
        <joint_name>pendulum_joint</joint_name>
        <topic_name>/apply_joint_effort</topic_name>
    </plugin>
    </gazebo>


</robot>